File: BookServiceJava/src/main/java/org/elcata98/mybooks/booksservice/model/Lending.java
package org.elcata98.mybooks.booksservice.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;
import org.apache.commons.lang3.builder.ToStringBuilder;
import org.apache.commons.lang3.builder.ToStringStyle;
import org.elcata98.mybooks.booksservice.util.LocalDateJsonDeserializer;
import org.elcata98.mybooks.booksservice.util.LocalDateJsonSerializer;
import org.springframework.data.mongodb.core.mapping.Document;

import javax.validation.constraints.NotNull;
import java.time.LocalDate;

@Document(collection = "lendings")
@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class Lending extends IdEntity {

    @NotNull
    private Book book;

    @NotNull
    private User who;

    @JsonSerialize(using = LocalDateJsonSerializer.class)
    @JsonDeserialize(using = LocalDateJsonDeserializer.class)
    private LocalDate startDate = LocalDate.now();

    @JsonSerialize(using = LocalDateJsonSerializer.class)
    @JsonDeserialize(using = LocalDateJsonDeserializer.class)
    private LocalDate endDate;

    public Book getBook() {
        return book;
    }

    public void setBook(Book book) {
        this.book = book;
    }

    public User getWho() {
        return who;
    }

    public void setWho(User who) {
        this.who = who;
    }

    public LocalDate getStartDate() {
        return startDate;
    }

    public void setStartDate(LocalDate startDate) {
        this.startDate = startDate;
    }

    public LocalDate getEndDate() {
        return endDate;
    }

    public void setEndDate(LocalDate endDate) {
        this.endDate = endDate;
    }

    @Override
    public int hashCode() {
        return HashCodeBuilder.reflectionHashCode(this, false);
    }

    @Override
    public boolean equals(Object obj) {
        return EqualsBuilder.reflectionEquals(this, obj, false);
    }

    @Override
    public String toString() {
        return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
    }
}

File: BooksService/src/main/kotlin/org/elcata98/mybooks/booksservice/model/Lending.kt
package org.elcata98.mybooks.booksservice.model

import org.apache.commons.lang3.builder.EqualsBuilder
import org.apache.commons.lang3.builder.HashCodeBuilder
import org.apache.commons.lang3.builder.ToStringBuilder
import org.apache.commons.lang3.builder.ToStringStyle
import java.util.Date
import java.util.UUID

class Lending {
    var lendingId: UUID = UUID.randomUUID()
        private set

    lateinit var who: User
    lateinit var book: Book
    lateinit var startDate: Date
    var endDate: Date? = null

    override fun equals(other: Any?): Boolean {

        return EqualsBuilder.reflectionEquals(this, other, false)
    }

    override fun hashCode(): Int {

        return HashCodeBuilder.reflectionHashCode(this, false)
    }

    override fun toString(): String {
        return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE)
    }
}